opsdroid/connector/matrix/connector.py

```diff
@@ -12,16 +12,17 @@
 
 from opsdroid import const, events
 from opsdroid.connector import Connector, register_event
-from voluptuous import Required
+from voluptuous import Required, Inclusive
 
 from . import events as matrixevents
 from .create_events import MatrixEventCreator
 from .html_cleaner import clean
 
 _LOGGER = logging.getLogger(__name__)
 CONFIG_SCHEMA = {
-    Required("mxid"): str,
-    Required("password"): str,
+    Inclusive("mxid", "login"): str,
+    Inclusive("password", "login"): str,
+    "access_token": str,
```

Comment:
@cognifloyd You mentioned something in chat about this. I think this is the best that can be done with what's supported in the lib at the moment?

Reply:
Yeah. I was saying we could add our own schema object to support requiring at least one auth method, either login or token.

The API might look something like:
```python
    Either("mxid", group="login", group_of_inclusion="userpass"): str,
    Either("password", group="login", group_of_inclusion="userpass"): str,
    Either("access_token", group="login", group_of_inclusion="token"): str,
```

Or maybe:
```python
    Either("mxid", "login.userpass"): str,
    Either("password", "login.userpass"): str,
    Either("access_token", "login.token"): str,
```

Reply:
I think I will leave this for a while.